packages/ui-components/src/components/select-multi-options/select-multi-options.tsx

```diff
@@ -63,8 +63,12 @@ export class KvSelectMultiOptions implements ISelectMultiOptionsConfig, ISelectM
 	};
 
 	private onOptionSelection = ({ detail: option }: CustomEvent<string>) => {
-		// triple bang for handling false, true and undefined
-		const newOptions = { ...this.selectedOptions, [option]: !!!this.selectedOptions?.[option] };
+		let { [option]: isSelected, ...newOptions } = this.selectedOptions;
+
+		if (!isSelected) {
```

Comment:
The `selectedOptions` is a `Record<string, boolean>` in which its value can be for example:
```typescript
{
    "option-a": true,
    "option-a": false
}
```

In case the user clicks on "option-a", its value won't be changed. I think I get what you are trying to achieve here, but probably you also need to change this prop type to an array instead of an object.

Reply:
I agree with you but this would imply a breaking change, can we move this discussion [here](https://github.com/kelvininc/ui-components/issues/311)?